core/src/main/java/io/grpc/ManagedChannelBuilder.java

```diff
@@ -75,6 +75,19 @@
   }
 
   /**
+   * Execute application code directly in the transport thread.
+   *
+   * <p>Depending on the underlying transport, using a direct executor may lead to substantial
+   * performance improvements. However, it also requires the application to not block under
+   * any circumstances.
+   *
+   * <p>Calling this method is semantically equivalent to calling {@link #executor(Executor)} and
+   * passing in a direct executor. However, this is the preferred way as it may allow the transport
+   * to perform special optimizations.
+   */
+  public abstract T directExecutor();
```

Comment:
@ejona86 

Probably worth a brief discussion about whether we think this interface is sufficient. The optimization in SerializeReentrantCallsDirectExecutor works just as well with a single threaded executor for example.

Alternatives to consider
- withExecutorType(ExecutorTypeEnum { DEFAULT, DIRECT, SINGLE-THREADED, ...}) 
- withThreads(int numthreads) - GRPC creates the executor itself, 0 implies direct etc.